packages/flutter/lib/src/material/drawer.dart

```diff
@@ -180,6 +185,9 @@ class DrawerController extends StatefulWidget {
   /// close the drawer.
   final DrawerAlignment alignment;
 
+  /// Optional callback that is called when a [Drawer] is opened or closed.
+  final DrawerCallback drawerCallback;
```

Comment:
maybe `onDrawerChange` or some such.

Or maybe even better, have two callbacks, `onOpen` and `onClose`.